src/material/material_properties.h

```diff
@@ -0,0 +1,383 @@
+
+  //! populates all Get-able material data from materials_
+  void PopulateData();
+
+  /*!
+    throws an exception if fissile_ids_ contains an unused material id or
+    if fissile_ids_ is empty in an eigen problem
+  */
+  void CheckFissileIDs() const;
+
+  /*!
+    throws an exception if n_material_ doesn't equal
+    the number of materials read in
+  */
+  void CheckNumberOfMaterials() const;
+
+  /*!
+    throws an exception if n_group_ doesn't equal
+    the number of groups in each material
+  */
+  void CheckNumberOfGroups() const;
+
+  //! calls checkValid on each material
+  void CheckValidEach() const;
+
+  //! calls CheckConsistent on materials_;
+  void CheckConsistent() const;
+
+  /*!
+    Reads file names listed in the parameter handler into an
+    unordered_map<int, std::string>
+  */
+  static std::unordered_map<int, std::string>
+  ReadMaterialFileNames(dealii::ParameterHandler& prm);
+
+  //! Reads fissile material ids listed in the parameter handler
+  static std::unordered_set<int>
+  ReadFissileIDs(dealii::ParameterHandler& prm);
+
+  /*!
+    parses Materials from the given file names
+    accepts both serialized and human-readable protobuf files
+    consistent with material.proto
+    can throw FailedToFindMaterialFile or FailedToParseMaterialFile
+  */
+  static std::unordered_map<int, Material>
+  ParseMaterials(const std::unordered_map<int, std::string>& file_name_map);
+
+  /*!
+    returns number, full name, abbreviation, and id of a material
+    as one self-describing string
+  */
+  static std::string
+  CombinedName(const std::pair<int, Material>& id_material_pair,
+               const std::string& delimiter = "|");
+
+  /*!
+    returns full name, abbreviation, and id of a material
+    as one self-describing string
+  */
+  static std::string
+  CombinedName(const Material& material,
+               const std::string& delimiter = "|");
+
+  /*!
+    extracts all vector properties from the Material
+    can throw MultipleDefinition exception
+  */
+  static std::unordered_map<Material::VectorId, std::vector<double>, std::hash<int>>
+  GetVectorProperties(const Material& material);
+
+  /*! 
+    returns the first instance of the specified vector property
+    found in the Material
+    will return an empty vector {} if the property isn't found
+  */
+  static std::vector<double>
+  GetVectorProperty(const Material& material, Material::VectorId property_id);
+
+  /*! 
+    returns the first instance of a SIGMA_S matrix found in the Material
+    will return an empty zero by zero matrix if SIGMA_S isn't found
+    can throw WrongNumberOfValues exception
+  */
+  static dealii::FullMatrix<double>
+  GetScatteringMatrix(const Material& material);
+
+ public:
+  /*
+    dealII macros for declaring exceptions are documented in
+    https://www.dealii.org/9.0.0/doxygen/deal.II/group__Exceptions.html
+    parameters are exception name,
+    argument data types,
+    and output sequence for print_info()
+  */
+
```

Comment:
One comment: if `dealii::` namespace is applicable to the macros, we should apply it (we don't if not)

One thing to change: add two more spaces for indent. Read Google Style Guide